File: frontend/src/api/hangman.ts
import { API_BASE_URL } from './config'
import type { HangmanWord } from '../types/hangman'

/** Fetch the authored word bank for the Hangman game. */
export async function fetchHangmanWords() {
  const response = await fetch(`${API_BASE_URL}/api/hangman/words`)
  if (!response.ok) {
    throw new Error('Nie udało się pobrać haseł wisielca.')
  }

  return (await response.json()) as HangmanWord[]
}

File: backend/app/api/hangman.py
from fastapi import APIRouter, HTTPException

from backend.app.core.hangman_word_loader import (
    HangmanWordLoaderError,
    load_hangman_words,
)
from backend.app.models.hangman import HangmanWord

router = APIRouter(prefix="/api/hangman", tags=["hangman"])


@router.get("/words", response_model=list[HangmanWord])
def list_hangman_words() -> list[HangmanWord]:
    """Return the authored word bank for the Hangman game."""

    try:
        return load_hangman_words()
    except HangmanWordLoaderError as error:
        raise HTTPException(status_code=500, detail=str(error)) from error
